core/src/test/scala/org/apache/spark/storage/MemoryStoreSuite.scala

```diff
@@ -546,4 +551,105 @@ class MemoryStoreSuite
       }
     }
   }
+
+  test("put native objects to MemoryStore and remove") {
```

Comment:
@srowen , I added two UnitTests to verify this PR, my initial purpose of adding this Manual close is to release memory allocated by Apache-Arrow, so below UT is based on Arrow, while using Arrow in spark-core UT requires an extra dependencies in core's pom, I am not sure if it is ok to add dependencies in pom? Or I'll try to write something else to make the point.